paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerNameEntityScriptEvent.java

```diff
@@ -0,0 +1,95 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.paper.PaperModule;
+import com.denizenscript.denizen.utilities.PaperAPITools;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import io.papermc.paper.event.player.PlayerNameEntityEvent;
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class PlayerNameEntityScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player names <entity>
+    //
+    // @Location true
+    //
+    // @Plugin Paper
+    //
+    // @Group Paper
+    //
+    // @Cancellable true
+    //
+    // @Triggers when a player attempts to rename an entity with a name tag.
+    //
+    // @Context
+    // <context.entity> returns an EntityTag of the renamed entity.
+    // <context.old_name> returns the old name of the entity, if any.
+    // <context.name> returns the new name of the entity.
+    // <context.persistent> returns whether this will cause the entity to persist through server restarts.
+    //
+    // @Determine
+    // "NAME:<ElementTag>" to set a different name for the entity.
+    // "PERSISTENT:<ElementTag(Boolean)>" to set whether the event will cause the entity to persist through restarts. NOTE: Entities may still persist for other reasons. To ensure they do not, use <@link mechanism EntityTag.force_no_persist>.
+    //
+    // @Player Always.
+    //
+    // -->
+
+    public PlayerNameEntityScriptEvent() {
+        registerCouldMatcher("player names <entity>");
+        this.<PlayerNameEntityScriptEvent, ElementTag>registerDetermination("persistent", ElementTag.class, (evt, context, determination) -> {
+            event.setPersistent(determination.asBoolean());
+        });
```

Comment:
This should probably be `registerOptionalDetermination` and return `false` and/or error if the element isn't a boolean?